graphdatascience/query_runner/cypher_graph_constructor.py

```diff
@@ -73,51 +71,244 @@ def _should_warn_about_arrow_missing(self) -> bool:
-        source_id_index = rel_columns.index("sourceNodeId")
-        target_id_index = rel_columns.index("targetNodeId")
-
-        type_query = ""
-        if "relationshipType" in rel_df.keys():
-            type_index = rel_columns.index("relationshipType")
-            type_query = f", relationship[{type_index}] as type"
-
-        property_query = ""
-        property_columns: Set[str] = set(rel_df.keys()) - {"sourceNodeId", "targetNodeId", "relationshipType"}
-        if len(property_columns) > 0:
-            property_queries = (f", relationship[{rel_columns.index(col)}] as {col}" for col in property_columns)
-            property_query = "".join(property_queries)
-
-        return (
-            "UNWIND $relationships as relationship "
-            f"RETURN relationship[{source_id_index}] as source, relationship[{target_id_index}] as target"
-            f"{type_query}{property_query}",
-            rel_list,
-        )
+    class CypherAggregationRunner:
+
+        _BIT_COL_SUFFIX = "_is_present" + str(uuid4())
+
+        def __init__(self, query_runner: QueryRunner, graph_name: str, concurrency: int):
+            self._query_runner = query_runner
+            self._concurrency = concurrency
+            self._graph_name = graph_name
+
+        def run(self, node_df: DataFrame, relationship_df: DataFrame) -> None:
+            graph_schema = self.schema(node_df, relationship_df)
+
+            same_cols = graph_schema.relationships.all.intersection(graph_schema.nodes.all)
+
+            if same_cols:
+                raise ValueError(
+                    "Expected disjoint column names in node and relationship df "
+                    f"but the columns {same_cols} exist in both dfs. Please rename the column in one df."
+                )
+
+            aligned_node_df = self.adjust_node_df(node_df, relationship_df, graph_schema)
+            aligned_rel_df = self.adjust_rel_df(relationship_df, node_df, graph_schema)
+
+            # concat instead of join as we want to first have all nodes and then the rels
+            # this way we dont duplicate the node property data and its cheaper
+            combined_df: DataFrame = concat([aligned_node_df, aligned_rel_df], ignore_index=True, copy=False)
+            # make column order deterministic
+            combined_df = combined_df[sorted(combined_df)]
+
+            # using a List and not a Set to preserve the order
+            combined_cols: List[str] = list(combined_df.columns)
+
+            nodes_config = self.nodes_config(combined_cols, graph_schema.nodes.all)
+            rels_config = self.rels_config(combined_cols, graph_schema.relationships.all)
+
+            property_clauses: List[str] = []
+
+            all_prop_cols = sorted(list(graph_schema.relationships.properties) + list(graph_schema.nodes.properties))
+
+            for prop_col in all_prop_cols:
+                property_clauses.append(self.check_value_clause(combined_cols, prop_col))
```

Comment:
maybe with a list-comprehension?

```suggestion
            all_prop_cols = sorted(list(graph_schema.relationships.properties) + list(graph_schema.nodes.properties))

            property_clauses = [self.check_value_clause(combined_cols, prop_col) for prop_col in all_prop_cols]
```